frontend/src/lib/validation.ts

```
@@ -0,0 +1,486 @@
+                        }
+                    }
+                    break;
+
+                case 'custom':
+                    if (rule.validator && !rule.validator(value as T[K] & string)) {
+                        return rule.message;
+                    }
+                    break;
+
+                case 'match':
+                    const matchValue = values[rule.value as keyof T];
```

Biome (2.3.13)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)